src/java/org/apache/cassandra/utils/btree/BTree.java

```diff
@@ -347,8 +347,8 @@ public static <Compare, Existing extends Compare, Insert extends Compare> Object
 
         if (isEmpty(toUpdate))
         {
-            if (isSimple(updateF))
-                return insert; // if update is empty and updateF is trivial, return our new input
+//            if (isSimple(updateF))
+//                return insert; // if update is empty and updateF is trivial, return our new input
```

Comment:
Can this be removed?

Reply:
Quite likely not. I just accidentally commeted it out during testing. Good catch!